### PR TITLE
feat: Move

### PR DESCRIPTION
**Pull request by cwhitten**
closes #2206 

https://share.getcloudapp.com/eDuxRoXn

### CODE REVIEW

File: Composer/packages/lib/shared/src/dialogFactory.ts [hunk 1]
```diff
@@ -146,7 +148,7 @@ export const seedNewDialog = (
       ...designerAttributes,
     },
     ...seedDefaults($type),
-    ...(initialDialogShape[$type] || {}),
+    ...(initialDialogShape[$type] ? initialDialogShape[$type](seededActions) : {}),
```
Comment (yeze322) [inline Composer/packages/lib/shared/src/dialogFactory.ts]: maybe we can check the type of initial shape, such as
```
if (typeof initialShape === 'function') {
  return initialShape(...params);
} else {
  return initialShape;
}
```
Then we don't need to wrap every type with arrow function

Comment (cwhitten) [reply Composer/packages/lib/shared/src/dialogFactory.ts]: ah i like that

**Comment by yeze322** [comment]
After moving SendActivty or Input types to new dialog, lg templates will be lost.
Relies on #2236 to fix that problem.

File: Composer/packages/client/src/pages/design/index.tsx [hunk 2]
```diff
@@ -312,8 +325,14 @@ function DesignPage(props) {
   }, [dialogs, breadcrumb]);
 
   async function onSubmit(data: { name: string; description: string }) {
-    const content = { ...getNewDesigner(data.name, data.description), generator: `${data.name}.lg` };
-    const seededContent = seedNewDialog('Microsoft.AdaptiveDialog', content.$designer, content);
+    const seededContent = seedNewDialog(
+      'Microsoft.AdaptiveDialog',
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/design/index.tsx]: nit: use `SDKTypes` enum here.

File: Composer/packages/client/src/pages/design/index.tsx [hunk 3]
```diff
@@ -312,8 +325,14 @@ function DesignPage(props) {
...
+      { name: data.name, description: data.description },
+      {
+        generator: `${data.name}.lg`,
+      }
+    );
+    set(seededContent, 'triggers[0].actions', state.actionsSeed || []);
```
Comment (a-b-r-o-w-n) [inline Composer/packages/client/src/pages/design/index.tsx]: Why not just add this in the `seedNewDialog` call?

File: Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts [hunk 4]
```diff
@@ -152,18 +152,22 @@ export function deleteNodes(inputDialog, nodeIds: string[], callbackOnRemovedDat
 }
 
 export function insert(inputDialog, path, position, $type) {
-  const dialog = cloneDeep(inputDialog);
-  const current = get(dialog, path, []);
   const newStep = {
     $type,
     ...seedNewDialog($type, { name: generateSDKTitle({ $type }) }),
```
Comment (a-b-r-o-w-n) [inline Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts]: ❓Should we move the name generation into `seedNewDialog`?

Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts]: I think so, we need a cleanup on all usages of seedNewDialog. Let me handle that in another PR.
image
image
@a-b-r-o-w-n do you think it's necessary to rename `seedNewDialog` to `seedAdaptiveElement` or some other names?

Comment (yeze322) [reply Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts]: https://github.com/microsoft/BotFramework-Composer/issues/2244

File: Composer/packages/client/src/pages/design/index.tsx (hunk 3 above)
Comment (yeze322) [reply Composer/packages/client/src/pages/design/index.tsx]: Chris had a solution that we can use function inside `initialDialogShape`
```ts
[SDKTypes.AdaptiveDialog]: (initialActions) => ({
    triggers: [
      { actions: initialActions }
    ]
})
```
I just noticed that it's lost after my commit, I will recover that

File: Composer/packages/client/src/pages/design/index.tsx (hunk 2 above)
Comment (yeze322) [reply Composer/packages/client/src/pages/design/index.tsx]: resolved

File: Composer/packages/client/src/pages/design/index.tsx (hunk 3 above)
Comment (yeze322) [reply Composer/packages/client/src/pages/design/index.tsx]: resolved

File: Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts (hunk 4 above)
Comment (a-b-r-o-w-n) [reply Composer/packages/extensions/visual-designer/src/utils/jsonTracker.ts]: Thanks @yeze322

**Pull request merged**